### PR TITLE
try to make arithmetic work in datavalue2

### PR DESCRIPTION
**Pull request by zhyass**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

## Changelog

- New Feature

## Related Issues

Fixes #issue

## Test Plan

Unit Tests

### CODE REVIEW

**Comment by zhyass** [comment]
CC @sundy-li , `cast` performance drop 10%～15%, `plus` performance drop 80%

Performance:

Before:
```
mysql> SELECT count(toUInt32(number)) FROM numbers(1000000000);
+-------------------------+
| count(toUInt32(number)) |
+-------------------------+
|              1000000000 |
+-------------------------+
1 row in set (0.19 sec)
Read 1000000000 rows, 8 GB in 0.186 sec., 5.39 billion rows/sec., 43.1 GB/sec.

mysql> SELECT count(toUInt32(number) + toInt64(number)) FROM numbers(1000000000);
+---------------------------------------------+
| count((toUInt32(number) + toInt64(number))) |
+---------------------------------------------+
|                                  1000000000 |
+---------------------------------------------+
1 row in set (0.43 sec)
Read 1000000000 rows, 8 GB in 0.420 sec., 2.38 billion rows/sec., 19.07 GB/sec.
```

After:
```
mysql> SELECT count(toUInt32(number)) FROM numbers(1000000000);
+-------------------------+
| count(toUInt32(number)) |
+-------------------------+
|              1000000000 |
+-------------------------+
1 row in set (0.22 sec)
Read 1000000000 rows, 8 GB in 0.216 sec., 4.64 billion rows/sec., 37.1 GB/sec.

mysql> SELECT count(toUInt32(number) + toInt64(number)) FROM numbers(1000000000);
+---------------------------------------------+
| count((toUInt32(number) + toInt64(number))) |
+---------------------------------------------+
|                                  1000000000 |
+---------------------------------------------+
1 row in set (2.21 sec)
Read 1000000000 rows, 8 GB in 2.203 sec., 454.02 million rows/sec., 3.63 GB/sec.
```

**Comment by sundy-li** [comment]
We can optimize the performance later, after we replace datavalues by datavalues2.

**Comment by zhyass** [comment]
TODO: interval arithmetic will be pushed in next pr.

File: common/functions/src/scalars/arithmetics/arithmetic_modulo.rs [hunk 1]
```diff
@@ -115,3 +63,85 @@ impl ArithmeticModuloFunction {
...
+    }
+
+    fn eval(&self, columns: &ColumnsWithField, _input_rows: usize) -> Result<ColumnRef> {
+        let lhs = columns[0].column();
+        let rhs = columns[1].column();
+
+        let left = ColumnViewerIter::<L>::try_create(lhs)?;
+        let right = ColumnViewerIter::<R>::try_create(rhs)?;
+
+        match (lhs.is_const(), rhs.is_const()) {
```
Comment (sundy-li) [inline common/functions/src/scalars/arithmetics/arithmetic_modulo.rs:101]: const,const case is already covered in Function2Adaptor?

**Pull request merged**